spec/build_spec.lua

```diff
@@ -451,3 +451,447 @@ describe("LuaRocks build tests #integration", function()
+            assert.truthy(build_builtin.run(rockspec))
+            local lib_file = "lib/luarocks/rocks-" .. test_env.lua_version .. "/c_module/1.0-1/lib/c_module"
+            if test_env.TEST_TARGET_OS == "windows" then
+               assert.truthy(lfs.attributes(lib_file .. ".dll"))
+            else
+               assert.truthy(lfs.attributes(lib_file .. ".so"))
+            end
+         end)
+         
+         it("returns true if the build succeeds with Lua module", function()
+            local rockspec = {
+               rockspec_format = "1.0",
+               package = "test",
+               version = "1.0-1",
+               source = {
+                  url = "http://example.com/test"
+               },
+               build = {
+                  type = "builtin",
+                  modules = {
+                     test = "test.lua"
+                  }
+               }
+            }
+            write_file("test.lua", "return {}", finally)
+            
+            rockspecs.from_persisted_table("test-1.0-1.rockspec", rockspec)
+            assert.truthy(build_builtin.run(rockspec))
+            assert.truthy(lfs.attributes("lib/luarocks/rocks-" .. test_env.lua_version .. "/test/1.0-1/lua/test.lua"))
+         end)
+         
+         it("automatically extracts the modules and libraries if they are not given and builds against any external dependencies", function()
+            local ssllib = "ssl"
+            if test_env.TEST_TARGET_OS == "windows" then
+               if test_env.MINGW then
+                  ssllib = "eay32"
+               else
+                  ssllib = "ssleay32"
+               end
+            end
+
+            local rockspec = {
+               rockspec_format = "3.0",
+               package = "c_module",
+               version = "1.0-1",
+               source = {
+                  url = "http://example.com/c_module"
+               },
+               external_dependencies = {
+                  openssl = {
+                     library = ssllib -- Use OpenSSL since it is available on all testing platforms
+                  }
+               },
+               build = {
+                  type = "builtin"
+               }
+            }
+            write_file("c_module.c", c_module_source, finally)
+            
+            rockspecs.from_persisted_table("c_module-1.0-1.rockspec", rockspec)
+            if test_env.TEST_TARGET_OS == "windows" then
+               local mingw_dir = ""
+               if test_env.MINGW then
+                  mingw_dir = "\\MinGW"
+               end
+               
+               rockspec.variables["openssl_INCDIR"] = "C:\\OpenSSL-Win32\\include" .. mingw_dir
+               rockspec.variables["openssl_LIBDIR"] = "C:\\OpenSSL-Win32\\lib" .. mingw_dir
+            else
+               rockspec.variables["openssl_INCDIR"] = "/usr/local/opt/openssl/include"
+               rockspec.variables["openssl_LIBDIR"] = "/usr/local/opt/openssl/lib"
```

Comment:
These locations (especially the Windows one) are already declared somewhere in `test_env.lua` (but I think they're declared in a single string, like `OPENSSL_INCDIR=... OPENSSL_LIBDIR=...`. Please edit that to create separate `test_env` variables and reuse them here (and in other places the OpenSSL libs are being used). Better not have hardcoded paths inside the test files that point to specific locations of the CI environment.